server/api/src/rsGetMiscSvrInfo.cpp

```diff
@@ -65,100 +65,104 @@ rsGetMiscSvrInfo( rsComm_t *rsComm, miscSvrInfo_t **outSvrInfo ) {
-                    GENERAL_NAME* gen_name = sk_GENERAL_NAME_value(gen_names, i);
-                    if(gen_name->type != GEN_DNS) {
-                        // TODO: support more than just dNSName
-                        continue;
-                    }
-                    int len = ASN1_STRING_to_UTF8(&tempStr, gen_name->d.dNSName);
-                    if(len > -1) {
-                        auto san_str = std::string(reinterpret_cast<char*>(tempStr), len);
-                        san_str.insert(0, "DNS:");
-                        san_vector.push_back(san_str);
+                // Set signature algorithm
+                certinfo_json["signature_algorithm"] = std::string(OBJ_nid2ln(X509_get_signature_nid(cert)));
+
+                // Set subject alternative name
+                GENERAL_NAMES* gen_names = static_cast<GENERAL_NAMES*>(X509_get_ext_d2i(cert, NID_subject_alt_name, nullptr, nullptr));
+                std::vector<std::string> san_vector;
+                if(gen_names) {
+                    unsigned char* tempStr;
+                    for(int i = 0; i < sk_GENERAL_NAME_num(gen_names); i++) {
+                        GENERAL_NAME* gen_name = sk_GENERAL_NAME_value(gen_names, i);
+                        if(gen_name->type != GEN_DNS) {
+                            // TODO: support more than just dNSName
+                            continue;
+                        }
+                        int len = ASN1_STRING_to_UTF8(&tempStr, gen_name->d.dNSName);
+                        if(len > -1) {
+                            auto san_str = std::string(reinterpret_cast<char*>(tempStr), len);
+                            san_str.insert(0, "DNS:");
+                            san_vector.push_back(san_str);
+                        }
                     }
                 }
+                if(san_vector.size() > 0) {
+                        certinfo_json["subject_alternative_names"] = san_vector;
+                }
             }
-            if(san_vector.size() > 0) {
-                    certinfo_json["subject_alternative_names"] = san_vector;
-            }
+        } else {
+            certinfo_json["ssl_enabled"] = false;
         }
-    } else {
-        certinfo_json["ssl_enabled"] = false;
-    }
 
-    const auto json_str = certinfo_json.dump();
-    myOutSvrInfo->certinfo.buf = malloc(json_str.size());
-    myOutSvrInfo->certinfo.len = json_str.size();
-    memcpy(myOutSvrInfo->certinfo.buf, json_str.c_str(), myOutSvrInfo->certinfo.len);
+        const auto json_str = certinfo_json.dump();
+        myOutSvrInfo->certinfo.buf = malloc(json_str.size());
+        myOutSvrInfo->certinfo.len = json_str.size();
+        memcpy(myOutSvrInfo->certinfo.buf, json_str.c_str(), myOutSvrInfo->certinfo.len);
+        } catch(const nlohmann::json::exception& e) {
+            irods::log(LOG_ERROR, fmt::format("[{}] Failed to serialize SSL/TLS data - {}", __FUNCTION__, e.what()));
```

Comment:
`__FUNCTION__` -> `__func__`

Prefer `__func__` since it's defined by the standard.